lib/SecretsPluginApi/extensionplugins.cpp

```diff
@@ -13,71 +13,1189 @@
+ * be provided by the user prior to use.
+ */
+bool PluginBase::unlock(const QByteArray &code)
 {
+    Q_UNUSED(code)
     return false;
 }
 
-bool PluginBase::setLockCode(const QByteArray &, const QByteArray &)
+/*!
+ * \brief Returns true if the lock code for the plugin was able to be set.
+ *
+ * The default implementation does nothing and returns false, as
+ * by default it is assumed that the plugin does not support locking.
+ * If the plugin implementation supports locking, and allows the lock
+ * code for the plugin to be changed, then this method should be overridden
+ * in order to perform the operation to set the lock code to the given
+ * \a newLockCode if the given \a oldLockCode was correct.
+ */
+bool PluginBase::setLockCode(const QByteArray &oldLockCode,
+                             const QByteArray &newLockCode)
 {
+    Q_UNUSED(oldLockCode)
+    Q_UNUSED(newLockCode)
     return false;
 }
 
+/*!
+  \class EncryptionPlugin
+  \brief Specifies an interface to derive an encryption key from
+         input data, and encrypt or decrypt secret data.
+
+  The EncryptionPlugin type specifies a simple interface which includes key
+  derivation, encryption and decryption operations.
+
+  The functionality provided by a concrete instance of this plugin is not
+  intended to be used directly by application clients, but instead is used to
+  encrypt (and decrypt) secret data which is stored in unencrypted storage
+  (that is, within a StoragePlugin).
+
+  Plugin implementers must be aware that the information reporting methods
+  (encryptionType(), and encryptionAlgorithm()) will be invoked from the main
+  thread of the secrets daemon, while the various interface operation methods
+  (deriveKeyFromCode(), encryptSecret(), and decryptSecret()) will be invoked
+  from a separate thread.  Plugins are loaded and plugin instances are
+  constructed in the main thread.
+ */
+
+/*!
+ * \brief Constructs a new EncryptionPlugin instance
+ */
 EncryptionPlugin::EncryptionPlugin()
     : PluginBase()
 {
 }
 
+/*!
+ * \brief Cleans up any memory associated with the EncryptionPlugin instance
+ */
 EncryptionPlugin::~EncryptionPlugin()
 {
 }
 
+/*!
+ * \enum EncryptionPlugin::EncryptionType
+ *
+ * This enum defines the types of encryption capability which may be offered by plugins
+ *
+ * \value NoEncryption No encryption is performed
+ * \value SoftwareEncryption Encryption is performed by "normal" rich execution environment application
+ * \value TrustedExecutionSoftwareEncryption Encryption is performed by trusted execution environment application
+ * \value SecurePeripheralEncryption Encryption is performed by a secure hardware peripheral via TEE application
+ */
+
+/*!
+ * \enum EncryptionPlugin::EncryptionAlgorithm
+ *
+ * This enum defines the encryption algorithms which may be used by plugins
+ *
+ * \value NoAlgorithm No encryption is performed
+ * \value CustomAlgorithm Some custom encryption algorithm is used by the plugin
```

Comment:
Thanks for this additional custom value ;)

Reply:
We should have a CustomWhatever in each enum.  If we don't that needs to be fixed - please let me know if you spot such a case.